File: 243.Shortest Word Distance/Jimmy Xiang/243.go
package main

import (
    "fmt"
)

func sortestDistance(word []string, word1 string, word2 string) int {
  for i, value := range  word {
         fmt.Println(i,value)
         fmt.Println(word1,word2)
    }

    return 3
}


func main() {

 a := []string{"a","b","c","d","f"}

 result := sortestDistance(a,"a","d")

 fmt.Println(result)

}

File: 243.Shortest Word Distance/Jimmy Xiang/243.py
class Solution:
    def sortestDistance(self, words, word1, word2):
        dist = float("inf")
        i, index1,index2 = 0, None, None
        while i < len(words):
            if words[i] == word1:
                index1 = i
            elif words[i] == word2:
                index2 = i
            if index1 is not None and index2 is not None:
                dist = min(dist, abs(index1 - index2))

            i += 1

        return dist



s=Solution()
a=["a","b","c","d"]
print s.sortestDistance(a,"a","d")


